docs/DeveloperGuide.adoc

```diff
@@ -137,256 +167,944 @@ image:BetterModelClassDiagram.png[]
+and `Income` transaction objects. For example, when inserting a new `Expense`/`Income`, the `AddTransactionCommandParser` will determine
+which object to initialize:
+
+.Sequence diagram of how adding a new `Transaction` is processed with polymorphism
+image::add-transaction/PolymorphismCalling.png[]
+
+`**Expense**` and `**Income**` are normally instantiated by either `ExpenseCommandParser#parse(String args)` or `IncomeCommandParser#parse(String args)`, which
+attempts to parse the various parameters supplied in `args` and return either a `**Expense**` or `**Income**` object. The following conditions will cause a `**ParseException**`
+to be thrown by the parser:
+
+. Missing parameters
+. Incorrect syntax (i.e. missing prefix, if it is required)
+. Illegal values in parameters (i.e. special characters and symbols entered for a integer-only field)
+. Multiple occurrences of parameters which only expects a single entry
+
+[NOTE]
+If the user input is incorrect due to any of the reasons above, the usage syntax will be shown.
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+Below is a given example on how a `**Transaction**` is added into `**THRIFT**` and how the back-end handles each step of the process:
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+Step 1. The user launches the application for the first time. THRIFT does not contain any other `**Transaction**` and the user is
+about to add their own. An empty transaction list is initialized.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+image::add-transaction/AddTransaction0.png[]
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+Step 2. The user executes **add_expense n/Laksa v/3.50** to insert an `**Expense**` with `**Description**` set to "Laksa"
+and `**Value**` set to "3.50". The `**TransactionDate**` is set to the user's current system date in the form "dd/mm/yyyy".
+The input is now checked and an attempt to parse each parameter occurs:
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+* `**Description**` is parsed by `AddTransactionCommandParser#parseTransactionDescription(ArgumentMultimap)`
```

Comment:
In my opinion, this part is unnecessarily detailed. You could probably just say that it's all parsed by various methods in `AddTransactionCommandParser`.